snuba/datasets/tags_column_processor.py

```diff
@@ -16,6 +27,39 @@
 NESTED_COL_EXPR_RE = re.compile(r"^([a-zA-Z0-9_\.]+)\[([a-zA-Z0-9_\.:-]+)\]$")
 
 
+@dataclass(frozen=True)
+class ParsedNestedColumn:
+    """
+    Provides some structure to the tags/contexts column to avoid parsing the string
+    in multiple places around the code.
+    """
+
+    col_name: str
+    parameter: Optional[str]
+
+    @classmethod
+    def parse_column_expression(cls, col_expr: str) -> Optional[ParsedNestedColumn]:
+        match = NESTED_COL_EXPR_RE.match(col_expr)
+        if match:
+            col_prefix = match[1]
+            param_string = match[2]
+
+            if col_prefix in ("tags", "contexts"):
```

Comment:
I suppose if this really is temporary this is fine, but this seems like it should use the same right hand side as line 60 (the body of `is_single_column`.)

I also don't really like how this is all so strongly coupled to the schema but I guess this was already how it was in `TagColumnProcessor` anyway.